### PR TITLE
[WIP]: PICARD-1172 : Common button for 'Add Files' and 'Add folder' 

### PR DESCRIPTION
**Pull request by vishichoudhary**
https://tickets.metabrainz.org/browse/PICARD-1172

### CODE REVIEW

File: picard/file_dilaog.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
+from PyQt5.QtGui import *
+from PyQt5.QtWidgets import *
+from PyQt5.QtCore import *
```
Comment (samj1912) [inline picard/file_dilaog.py]: Never make start imports. Its better to be verbose about where we are getting our classes and variables from.

File: picard/file_dilaog.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+import os,sys
+if os.name == 'nt':
```
Comment (samj1912) [inline picard/file_dilaog.py]: Is this correct? why not os.name == 'win32' like in other places in code.

File: picard/file_dilaog.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    import win32api, win32con
+from datetime import datetime
+
+def Is_hidden(file_temp):
+    if os.name== 'nt':
+        attribute = win32api.GetFileAttributes(file_temp)
```
Comment (samj1912) [inline picard/file_dilaog.py]: Why are you using the win32 api? You can instead subclass QFileDialog and use Qt implemented methods. see http://doc.qt.io/qt-5/qfiledialog.html#Option-enum

File: picard/file_dilaog.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        pic = QPixmap(imagePath)
+        self.setPixmap(pic)
+
+class File_dialog(QDialog):
+    """
+    Main class used for the dialog
+    """
+    def __init__(self,wi,he,path,parent=None):
+        self.path = path
+        self.formats=["MP3","FLAC","OGG","M4A","WMA","WAV","MP4","AAC","flv"]
```
Comment (samj1912) [inline picard/file_dilaog.py]: You should not hardcode this list. Instead import it from picard.formats

File: picard/file_dilaog.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        self.selectBtn.clicked.connect(lambda:self.select_action(True))
+
+        self.selectBtn1 = QPushButton("Open Folder",self)
+        self.selectBtn1.resize(150,30)
+        self.selectBtn1.move(wi-.219*wi,10)
+        self.selectBtn1.clicked.connect(lambda:self.open_action(True))
+
+        self.lineedit = QLineEdit("Search here",self)
+        self.lineedit.resize(250,30)
+        self.lineedit.move(wi-.51*wi,10)
```
Comment (samj1912) [inline picard/file_dilaog.py]: Please do not set such arbitrary numbers, instead make a .ui file via Qt designer and convert it to _ui file via `python setup.py build_ui` 
I would prefer if you separate the UI and the logic parts like the rest of the UI components do.

File: picard/file_dilaog.py [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+
+        self.table = QTableWidget(self)
+        self.table.setColumnCount(3)
+        self.table.setHorizontalHeaderLabels(("Name;Size;Modified").split(";"))
+        self.table.verticalHeader().setVisible(False)
+        self.table.setEditTriggers(QAbstractItemView.NoEditTriggers)
+        self.table.setSelectionBehavior(QAbstractItemView.SelectRows)
+        cw=int(.7*self.l1w)
+        cw1=int(.143*self.l1w)
+        cw2=int(.143*self.l1w)
```
Comment (samj1912) [inline picard/file_dilaog.py]: Avoid such arbitrary variable names. Use something verbose

File: picard/file_dilaog.py [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+        self.show_files(False)
+        self.table.resize(self.l1w,he-.26*he)
+        self.table.move(self.lw,50)
+        self.table.setShowGrid(False)
+        self.table.doubleClicked.connect(self.default_setting)
+
+        self.checkBox = QCheckBox("Show hidden files/folders",self)
+        self.checkBox.resize(200,30)
+        self.checkBox.move(wi-.365*wi,he-.17*he)
+        self.checkBox.toggled.connect(lambda:self.show_hidden(self.checkBox))
```
Comment (samj1912) [inline picard/file_dilaog.py]: Use `partial` here.

File: picard/file_dilaog.py [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+        sys.exit(self.exec_())
+
+    def format_modifier(self,i):
+
+        self.formats.clear()
+        self.formats_lower.clear()
+        if not self.cb.itemText(i) == "All Supported Formats":
+            self.formats = [self.cb.itemText(i)]
+        else:
+            self.formats=["MP3","FLAC","OGG","M4A","WMA","WAV","MP4","AAC","flv"]
```
Comment (samj1912) [inline picard/file_dilaog.py]: ditto as previous comment about formats.

File: picard/file_dilaog.py [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+
+        for i in self.formats:self.formats_lower.append(i.lower())
+
+        self.show_files(False)
+
+    def default_setting(self):
+        indexes = self.table.selectionModel().selectedRows()
+        if len(indexes) == 1:
+            #for index in (indexes):
+            if os.path.isdir(os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])):
```
Comment (samj1912) [inline picard/file_dilaog.py]: explain this better. I have 0 idea what this does.

File: picard/file_dilaog.py [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+            self.formats=["MP3","FLAC","OGG","M4A","WMA","WAV","MP4","AAC","flv"]
+
+        for i in self.formats:self.formats_lower.append(i.lower())
+
+        self.show_files(False)
+
+    def default_setting(self):
+        indexes = self.table.selectionModel().selectedRows()
+        if len(indexes) == 1:
+            #for index in (indexes):
```
Comment (samj1912) [inline picard/file_dilaog.py]: Remove leftover code.

File: picard/file_dilaog.py [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+        self.send_things = []
+        for index in (indexes):
+            hola = (self.table.item(index.row(),0)).text()[6:]
+            if os.path.isdir(os.path.join(self.path,hola)):
+                self.recu_folder(self.send_things,os.path.join(self.path,hola))
+            else:
+                self.send_things.append(os.path.join(self.path,hola))
+        for  i in self.send_things:
+            print(i)
+        #print(self.send_things)
```
Comment (samj1912) [inline picard/file_dilaog.py]: left over code.

File: picard/file_dilaog.py [hunk 12]
```diff
@@ -0,0 +1,274 @@
...
+            indexes = self.table.selectionModel().selectedRows()
+        self.send_things = []
+        for index in (indexes):
+            hola = (self.table.item(index.row(),0)).text()[6:]
+            if os.path.isdir(os.path.join(self.path,hola)):
+                self.recu_folder(self.send_things,os.path.join(self.path,hola))
+            else:
+                self.send_things.append(os.path.join(self.path,hola))
+        for  i in self.send_things:
+            print(i)
```
Comment (samj1912) [inline picard/file_dilaog.py]: use the logger instead of print.

File: picard/file_dilaog.py [hunk 13]
```diff
@@ -0,0 +1,274 @@
...
+                hola = (self.table.item(indexes[0].row(),0)).text()[6:]
+                self.path = os.path.join(self.path,hola)
+                self.show_files(False)
+
+    def select_action(self,indexes):
+        if indexes == True:
+            indexes = self.table.selectionModel().selectedRows()
+        self.send_things = []
+        for index in (indexes):
+            hola = (self.table.item(index.row(),0)).text()[6:]
```
Comment (samj1912) [inline picard/file_dilaog.py]: again, random variable names?

File: picard/file_dilaog.py [hunk 14]
```diff
@@ -0,0 +1,274 @@
...
+            if os.path.isdir(os.path.join(self.path,hola)):
+                self.recu_folder(self.send_things,os.path.join(self.path,hola))
+            else:
+                self.send_things.append(os.path.join(self.path,hola))
+        for  i in self.send_things:
+            print(i)
+        #print(self.send_things)
+        sys.exit()
+
+    def recu_folder(self,lis,path):
```
Comment (samj1912) [inline picard/file_dilaog.py]: please make sure you method names and variables names are appropriate. use something like `add_folder_recursive`

File: picard/file_dilaog.py [hunk 15]
```diff
@@ -0,0 +1,274 @@
...
+       temp = os.listdir(path)
+       for i in temp:
+            if os.path.isfile(os.path.join(path,i)):
+                for j in self.formats_lower:
+                    if i.endswith('.'+j):
+                        lis.append(os.path.join(path,i))
+            else:
+                self.recu_folder(lis,os.path.join(path,i))
+
+    def moveBack_action(self):
```
Comment (samj1912) [inline picard/file_dilaog.py]: Stick to a single naming convention. In picard we follow snake case.

File: picard/file_dilaog.py [hunk 16]
```diff
@@ -0,0 +1,274 @@
...
+        self.hidden_items = 0
+        self.files_to_show()
+        if hidden == False :
+            self.table.setRowCount(len(self.dirs)+len(self.files)-self.hidden_items)
+        else:
+            self.table.setRowCount(len(self.dirs)+len(self.files))
+        start,pos,j=0,0,0
+        while start < len(self.dirs):
+            if hidden == False :
+                if Is_hidden(self.dirs[start]) == True:
```
Comment (samj1912) [inline picard/file_dilaog.py]: Again, naming convention with Is_hidden, why the captial `I`?

File: picard/file_dilaog.py [hunk 17]
```diff
@@ -0,0 +1,274 @@
...
+                    start=start+1
+                    continue
+            self.table.setItem(pos,0,QTableWidgetItem("      "+self.dirs[start]))
+            self.table.setItem(pos,1,QTableWidgetItem(self.dirs_size[start]))
+            self.table.setItem(pos,2,QTableWidgetItem(self.dirs_modi_time[start]))
+            self.table.setCellWidget(pos, 0, ImgWidget1("../resources/img-src/folder24_24.png"))
+            pos,start=pos+1,start+1
+
+        while j < len(self.files):
```
Comment (samj1912) [inline picard/file_dilaog.py]: Do no use arbitrary iterator variables in case they mean something. All I see you doing here is calling a function with file positions, it might be better to first filter out all the hidden files and then implement your logic. This seems closer to competitive coding then application coding. Make your logic as clear as possible so that it is easier to follow.

File: picard/file_dilaog.py [hunk 18]
```diff
@@ -0,0 +1,274 @@
...
+            self.show_files(False)
+
+    def files_to_show(self):
+        self.files = []
+        self.files_size = []
+        self.files_modi_time = []
+        self.dirs = []
+        self.dirs_size = []
+        self.dirs_modi_time = []
+        def converter(num):
```
Comment (samj1912) [inline picard/file_dilaog.py]: Move this to picard.utils

File: picard/file_dilaog.py [hunk 19]
```diff
@@ -0,0 +1,274 @@
...
+                self.dirs_modi_time.append(date_)
+            else:
+                for j in self.formats_lower:
+                    if i.endswith('.'+j) == True:
+                        self.files.append(i)
+                        self.files_size.append(converter(file_info.st_size))
+                        self.files_modi_time.append(date_)
+                        break
+
+if __name__ == '__main__':
```
Comment (samj1912) [inline picard/file_dilaog.py]: Why are you checking for a script execution? Instead fix the dialog to fire on add button calls once picard is open.
Remove all the code below.

**Comment by samj1912** [comment]
Next time break your PR into smaller parts. This was really hard to review. Also move the file inside the ui folder and follow Picard conventions. You might want to read the contribution guidelines.https://github.com/metabrainz/picard/blob/master/CONTRIBUTING.md

File: picard/file_dilaog.py (hunk 4 above)
Comment (zas) [reply picard/file_dilaog.py]: Also please respect PEP8 ! Read 
  https://github.com/metabrainz/picard/blob/master/CONTRIBUTING.md

File: picard/file_dilaog.py (hunk 6 above)
Comment (zas) [reply picard/file_dilaog.py]: Variable names should be self-explanatory, those are clearly not.

File: picard/file_dilaog.py [hunk 20]
```diff
@@ -0,0 +1,274 @@
...
+        self.table.setColumnWidth(2,cw2)
+        self.show_files(False)
+        self.table.resize(self.l1w,he-.26*he)
+        self.table.move(self.lw,50)
+        self.table.setShowGrid(False)
+        self.table.doubleClicked.connect(self.default_setting)
+
+        self.checkBox = QCheckBox("Show hidden files/folders",self)
+        self.checkBox.resize(200,30)
+        self.checkBox.move(wi-.365*wi,he-.17*he)
```
Comment (zas) [inline picard/file_dilaog.py]: Why is this needed ?

File: picard/file_dilaog.py (hunk 8 above)
Comment (zas) [inline picard/file_dilaog.py]: Be consistent and respect PEP 8: `self.formats = [...]` is correct, `self.formats=[...]` isn't.

File: picard/file_dilaog.py [hunk 21]
```diff
@@ -0,0 +1,274 @@
...
+                self.select_action(indexes)
+        else :
+            self.select_action(indexes)
+
+    def open_action(self,indexes):
+        if indexes == True:
+            indexes = self.table.selectionModel().selectedRows()
+        if len(indexes) == 1:
+            if os.path.isdir(os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])):
+                hola = (self.table.item(indexes[0].row(),0)).text()[6:]
```
Comment (zas) [inline picard/file_dilaog.py]: What is `hola` ??

File: picard/file_dilaog.py [hunk 22]
```diff
@@ -0,0 +1,274 @@
...
+
+        while j < len(self.files):
+            if hidden == False :
+                if Is_hidden(self.files[j]) == True:
+                    j=j+1
+                    continue
+            self.table.setItem(pos,0,QTableWidgetItem("      "+self.files[j]))
+            self.table.setItem(pos,1,QTableWidgetItem(self.files_size[j]))
+            self.table.setItem(pos,2,QTableWidgetItem(self.files_modi_time[j]))
+            self.table.setCellWidget(pos, 0, ImgWidget1("../resources/img-src/music24_24.png"))
```
Comment (samj1912) [inline picard/file_dilaog.py]: Compile your resources properly. We use a qrc file to get icons. See picard/resources on how to do this.

File: picard/file_dilaog.py [hunk 23]
```diff
@@ -0,0 +1,274 @@
...
+
+    def moveBack_action(self):
+        self.path = os.path.dirname((os.path.abspath(self.path)))
+        self.show_files(False)
+
+    def moveNext_action(self):
+        indexes = self.table.selectionModel().selectedRows()
+        if len(indexes) == 1:
+            if os.path.isdir(os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])):
+                self.path = os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])
```
Comment (zas) [inline picard/file_dilaog.py]: Too much code redundancy, cache `os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])` (and explain it...)

File: picard/file_dilaog.py [hunk 24]
```diff
@@ -0,0 +1,274 @@
...
+                self.show_files(False)
+        if len(indexes) == 0:
```
Comment (zas) [inline picard/file_dilaog.py]: `if not indexes:` is more pythonic.

File: picard/file_dilaog.py [hunk 25]
```diff
@@ -0,0 +1,274 @@
...
+            temp_name = []
+            for i in os.listdir(self.path) :
+                if os.path.isdir(os.path.join(self.path,i)):
+                    temp_name.append(i)
+            if len(temp_name) >0:
+                #sort(temp_name)
```
Comment (zas) [inline picard/file_dilaog.py]: Remove commented out code.

File: picard/file_dilaog.py [hunk 26]
```diff
@@ -0,0 +1,274 @@
...
+                self.path=os.path.join(self.path,temp_name[0])
+                self.show_files(False)
+
+    def show_files(self,hidden):
+        self.hidden_items = 0
+        self.files_to_show()
+        if hidden == False :
```
Comment (zas) [inline picard/file_dilaog.py]: `if not hidden:` is more pythonic.

File: picard/file_dilaog.py (hunk 16 above)
Comment (zas) [reply picard/file_dilaog.py]: `if is_hidden(...): ` should be enough.

File: picard/file_dilaog.py (hunk 18 above)
Comment (zas) [reply picard/file_dilaog.py]: Just re-use https://github.com/metabrainz/picard/blob/master/picard/util/bytes2human.py

File: picard/file_dilaog.py [hunk 27]
```diff
@@ -0,0 +1,274 @@
...
+                else:
+                    self.hidden_items=self.hidden_items+1
+            date_= str((datetime.fromtimestamp(file_info.st_mtime)).date())
+            if os.path.isdir(os.path.join(self.path,i)):
+                self.dirs.append(i)
+                self.dirs_size.append("NA")
+                self.dirs_modi_time.append(date_)
+            else:
+                for j in self.formats_lower:
+                    if i.endswith('.'+j) == True:
```
Comment (zas) [inline picard/file_dilaog.py]: What's `j` here ?
Also, in Python, we don't usually test against False / True: `if i.endswith('.' + j):`

File: picard/file_dilaog.py [hunk 28]
```diff
@@ -0,0 +1,274 @@
...
+        return file_temp.startswith('.')
+
+class ImgWidget1(QLabel):
+    """
+    Class used to show picture on Table Row
+    """
+    def __init__(self, imagePath,parent=None):
+        super(ImgWidget1, self).__init__(parent)
+        pic = QPixmap(imagePath)
+        self.setPixmap(pic)
```
Comment (zas) [inline picard/file_dilaog.py]: `pic` intermediate variable is useless here.

File: picard/file_dilaog.py [hunk 29]
```diff
@@ -0,0 +1,274 @@
...
+
+class File_dialog(QDialog):
+    """
+    Main class used for the dialog
+    """
+    def __init__(self,wi,he,path,parent=None):
+        self.path = path
+        self.formats=["MP3","FLAC","OGG","M4A","WMA","WAV","MP4","AAC","flv"]
+        self.formats_lower=[]
+        for i in self.formats:self.formats_lower.append(i.lower())
```
Comment (zas) [inline picard/file_dilaog.py]: Very bad style.

Use a list comprehension for the initialization instead:

```python
self.formats_lower = [ format.lower() for format in self.formats]
```

Is this list really needed ?

File: picard/file_dilaog.py [hunk 30]
```diff
@@ -0,0 +1,274 @@
...
+        self.formats_show=["MP3","FLAC","OGG","M4A","WMA","WAV","MP4","AAC","flv"]
```
Comment (zas) [inline picard/file_dilaog.py]: Why are there `formats` and `formats_show` ?
If you really need a label & a value, keep them associated using a list of objects, instead of multiple independent lists.

File: picard/file_dilaog.py (hunk 9 above)
Comment (vishichoudhary) [reply picard/file_dilaog.py]: Ok i will add comments for such codes. Basically on single click it will open a folder and if a file is selected it will select that file.

File: picard/file_dilaog.py [hunk 31]
```diff
@@ -0,0 +1,274 @@
...
+                    j=j+1
+                    continue
+            self.table.setItem(pos,0,QTableWidgetItem("      "+self.files[j]))
+            self.table.setItem(pos,1,QTableWidgetItem(self.files_size[j]))
+            self.table.setItem(pos,2,QTableWidgetItem(self.files_modi_time[j]))
+            self.table.setCellWidget(pos, 0, ImgWidget1("../resources/img-src/music24_24.png"))
+            pos,j=pos+1,j+1
+
+    def search_update(self):
+        print(self.lineedit.text())
```
Comment (zas) [inline picard/file_dilaog.py]: Spurious `print()`

File: picard/file_dilaog.py [hunk 32]
```diff
@@ -0,0 +1,274 @@
...
+        self.lineedit.setText("")
+
+    def show_hidden(self,cb):
+        if cb.isChecked() == True:
```
Comment (zas) [inline picard/file_dilaog.py]: Shorter: `self.show_files(cb.isChecked())`

**Comment by vishichoudhary** [comment]
Getting following error 
Traceback (most recent call last):
  File "tagger.py", line 7, in <module>
    from picard.tagger import main
  File "./picard/tagger.py", line 69, in <module>
    from picard.ui.mainwindow import MainWindow
  File "./picard/ui/mainwindow.py", line 40, in <module>
    from picard.ui.filedialog import FileDialog
  File "./picard/ui/filedialog.py", line 22, in <module>
    fi = FileDialog()
  File "./picard/ui/filedialog.py", line 12, in __init__
    self.ui.setupUi(self)
  File "./picard/ui/ui_filedialog.py", line 64, in setupUi
    self.retranslateUi(fileDialog)
  File "./picard/ui/ui_filedialog.py", line 69, in retranslateUi
    print(_)
NameError: name '_' is not defined

File: picard/ui/filedialog.py [hunk 33]
```diff
@@ -0,0 +1,77 @@
+from picard.ui import PicardDialog
+from picard.ui.ui_filedialog import Ui_fileDialog
+from picard.formats import supported_extensions
+from PyQt5.QtWidgets import (
+        QFileSystemModel , QAbstractItemView ,
+        QListWidgetItem)
+from PyQt5.QtCore import QStorageInfo, QDir, QSortFilterProxyModel , QRegExp
+import os,sys
```
Comment (samj1912) [inline picard/ui/filedialog.py]: Please import only one module at a time. Also order your imports. See https://www.python.org/dev/peps/pep-0008/#imports

Comment (samj1912) [reply picard/ui/filedialog.py]: Also see the codacy issues - https://www.codacy.com/app/MetaBrainz/picard/pullRequest?prid=1214235

File: picard/ui/filedialog.py [hunk 34]
```diff
@@ -0,0 +1,77 @@
...
+        self.ui.formatsCb.currentIndexChanged.connect(self.change_extension)
+        self.ui.selectBtn.clicked.connect(self.get_files)
+
+    def closeEvent(self,event):
+        self.files.clear()
+        self.close()
+
+    def get_files(self):
+        indexes = self.ui.fileTree.selectionModel().selectedIndexes()
+        for indx in indexes:
```
Comment (zas) [inline picard/ui/filedialog.py]: `indexes` is useless here, and please cache repeated function calls return value:

```python
for indx in self.ui.fileTree.selectionModel().selectedIndexes():
    path = self.model.filePath(indx)
    if os.path.isdir(path):
       self.add_directory_recursively(path)
    else:
       self.files.append(path)
```

File: picard/ui/filedialog.py [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+            if os.path.isdir(self.model.filePath(indx)):
+                self.add_directory_recursively(self.model.filePath(indx))
+            else:
+                self.files.append(self.model.filePath(indx))
+        self.hide()
+        return self.files
+
+    def add_directory_recursively(self,path):
```
Comment (zas) [inline picard/ui/filedialog.py]: PEP8: add a space after the comma ^^

File: picard/ui/filedialog.py [hunk 36]
```diff
@@ -0,0 +1,77 @@
...
+        for file in os.listdir(path):
```
Comment (zas) [inline picard/ui/filedialog.py]: Again, please prevent multiple useless function calls in loops
```python
fullpath = os.path.join(path, file)
...
```
and replace 4 occurences of `os.path.join(path, file)`

**Comment by samj1912** [comment]
This doesn't seem to solve our unification woes. It is an exact copy of the FileBrowser pane of Picard. What we need is a ListView implementation of adding files, not TreeView.

File: picard/ui/filedialog.py [hunk 37]
```diff
@@ -0,0 +1,77 @@
...
+        self.model.setNameFilterDisables(0)
+        self.ui.fileTree.setModel(self.model)
+        self.ui.fileTree.setCurrentIndex(self.model.index(QDir.currentPath())) #temporary
+        self.ui.fileTree.setColumnWidth(0,.7*self.ui.fileTree.width())
+        self.ui.fileTree.setColumnWidth(1,.1*self.ui.fileTree.width())
+        self.ui.fileTree.setColumnWidth(2,.1*self.ui.fileTree.width())
+        self.ui.fileTree.setColumnWidth(3,.1*self.ui.fileTree.width())
+        self.ui.fileTree.setSelectionMode(QAbstractItemView.ExtendedSelection)
+
+        self.ui.formatsCb.addItem('All Supported Formats')
```
Comment (zas) [inline picard/ui/filedialog.py]: I18N ?

File: picard/ui/filedialog.py [hunk 38]
```diff
@@ -0,0 +1,77 @@
...
+            if os.path.isfile(os.path.join(path,file)):
+                for exts in supported_extensions():
+                    if file.endswith(exts):
+                            self.files.append(os.path.join(path,file))
+            elif os.path.isdir(os.path.join(path,file)):
+                    self.add_directory_recursively(os.path.join(path,file))
+
+    def change_extension(self,ext):
+        self.extensions.clear()
+        if self.ui.formatsCb.itemText(ext) == 'All Supported Formats':
```
Comment (zas) [inline picard/ui/filedialog.py]: Not sure to understand, but this string is translatable right ?

File: picard/ui/filedialog.py [hunk 39]
```diff
@@ -0,0 +1,77 @@
...
+            self.extensions = ['*' + ext for ext in supported_extensions()]
+        else:
+            self.extensions.append('*' + (self.ui.formatsCb.itemText(ext)))
```
Comment (zas) [inline picard/ui/filedialog.py]: Again, cache `self.ui.formatsCb.itemText(ext)` (and remove useless extra parenthesis).

File: picard/ui/mainwindow.py [hunk 40]
```diff
@@ -729,7 +730,8 @@ def add_files(self):
         formats.sort()
         extensions.sort()
         formats.insert(0, _("All Supported Formats") + " (%s)" % " ".join(extensions))
-        files, _filter = QtWidgets.QFileDialog.getOpenFileNames(self, "", current_directory, ";;".join(formats))
+        #files, _filter = QtWidgets.QFileDialog.getOpenFileNames(self, "", current_directory, ";;".join(formats))
```
Comment (zas) [inline picard/ui/mainwindow.py]: This commented line needs to be removed in final patch.

File: picard/ui/filedialog.py (hunk 34 above)
Comment (vishichoudhary) [reply picard/ui/filedialog.py]: ok, i will take care of it.

File: picard/ui/filedialog.py (hunk 35 above)
Comment (vishichoudhary) [reply picard/ui/filedialog.py]: ok i needs to adapt this .

File: picard/ui/mainwindow.py (hunk 40 above)
Comment (vishichoudhary) [reply picard/ui/mainwindow.py]: Yeah these lines will not be there in final patch.

**Comment by johnfoxii** [comment]
how about this?

**Comment by sambhav** [comment]
This has been a WIP for more than a month. I'm closing this for now. You can reopen it later when you want to work on it.
